### DISCUSSION TITLE
Release/5.1.3

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1224] [Announcement]
* We now test against, and package with, Python 3.9.6.

  **As a consequence of this we no longer support Windows 7.
  This is due to [Python 3.9.x itself not supporting Windows 7](https://www.python.org/downloads/windows/).
  The application (both EDMarketConnector.exe and EDMC.exe) will crash on startup due to a missing DLL.**

  This should have no other impact on users or plugin developers, other than the latter now being free to use features that were introduced since the Python 3.7 series.

  Developers can check the contents of the `.python-version` file in the source (it's not distributed with the Windows installer) for the currently used version in a given branch.

Release 5.1.3
===

* Attempt to flush any pending EDSM API data when a Journal `Shutdown` or `Fileheader` event is seen.  After this, the data is dropped.  This ensures that, if the user next logs in to a different commander, the data isn't then sent to the wrong EDSM account.

* Ensure a previous Journal file is fully read/drained before starting processing of a new one.  In particular, this ensures properly seeing the end of a continued Journal file when opening the continuation file.

* New config options, in a new `Privacy` tab, to hide the current Private Group, or captain of a ship you're multi-crewing on.  These usually appear on the `Commander` line of the main UI, appended after your commander name, with a `/` between.

* EDO dockable settlement names with `+` characters appended will no longer cause 'server lagging' reports.

* Don't force DEBUG level logging to the [plain log file](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#plain-log-file) if `--trace` isn't used to force TRACE level logging.  This means logging *to the plain log file* will once more respect the user-set Log Level, as in the Configuration tab of Settings.

  As its name implies, the [debug log file](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#debug-log-files) will always contain at least DEBUG level logging, or TRACE if forced.

(Plugin) Developers
---

* New EDMarketConnector option `--trace-on ...` to control if certain TRACE level logging is used or not.  This helps keep the noise down whilst being able to have users activate choice bits of logging to help track down bugs.

  See [Contributing.md](Contributing.md#use-the-appropriate-logging-level) for details.

* Loading of `ShipLocker.json` content is now tried up to 5 times, 10ms apart, if there is a file loading, or JSON decoding, failure.  This should hopefully result in the data being loaded correctly if a race condition with the game client actually writing to and closing the file is encountered.

* `config.get_bool('some_str', default=SomeDefault)` will now actually honour that specified default.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.1.3'>Release/5.1.3</a>.</em>

**Comment by sgtgaryk** [comment]
Athanasius, wanted to make you aware that Windows Threat Protection is flagging a threat in the .msi file upon download - whether real or not, it's popping up for me in the latest release. Screen captured the output of the alert & attached
![Screenshot 2021-08-13 192620](https://user-images.githubusercontent.com/67621525/129428935-c50b15ff-3218-49b6-908a-e60de7b759fa.jpg)

**Reply by Athanasius** [reply]
Thanks, I've reported it to them as a false positive.

If you look at the detail it's only triggering on EDMC.exe, so even if you believed the report it's still safe to use the GUI EDMarketConnector.exe.

Be aware that if you let the file be quarantined then you might run into issues installing a later version, with the AV blocking updating of the file.

**Reply by Athanasius** [reply]
> Analyst comments:
> 
> 
> We have removed the detection.  Please follow the steps below to clear cached detection and obtain the latest malware definitions.
> 
> 1. Open command prompt as administrator and change directory to c:\Program Files\Windows Defender 
> 2. Run “MpCmdRun.exe -removedefinitions -dynamicsignatures”
> 3. Run "MpCmdRun.exe -SignatureUpdate"
> 
> Alternatively, the latest definition is available for download here: https://www.microsoft.com/en-us/wdsi/definitions
> 
> Thank you for contacting Microsoft.

**Comment by sgtgaryk** [comment]
Thanks for the reply! I was relatively sure it was a false positive but wanted to make sure you knew about it. Hopefully I can get it unquarantined and running again.

Gary

________________________________

**Comment by Elryzor** [comment]
I'm having the same issue. I followed the instructions from Microsoft and I'm still seeing the same Trojan warning. This has only been happening since late last week - August 16 or so. I switched to a new computer then.

**Reply by Athanasius** [reply]
> I'm having the same issue. I followed the instructions from Microsoft and I'm still seeing the same Trojan warning. This has only been happening since late last week - August 16 or so. I switched to a new computer then.

I just ran a fresh scan on my `5.1.3` install (specifically on its install folder) and nothing was found by Defender.  I have:

```
Security intelligence version: 1.347.88.0
Version created on: 20/08/2021 07:30
Last updated: 20/08/2021 14:12
```

There's nothing still listed in my `Allowed threats`.  I do have a 'Remediation incomplete' still in the history for the 5.1.3 installer due to the EDMC.exe.  But, as I said, that's now scanning clean.

**Reply by Athanasius** [reply]
I should note that it was the edmc.exe that I'd submitted and gotten cleared.

Silly MicroSoft decided not to have that *also* clear the MSI that contains that specific EDMC.exe.  But someone else submitted it and that should now also be cleared.

**Reply by Elryzor** [reply]
Thank you for being so quick to respond to this.

On my computer, Windows now (Aug 21 2021, 10:00 AM EDT) reports no issues with the 5.1.3 .msi file. 

It was Aug 14th when this was first detected on my computer rather than Aug 16th as I originally said.

**Comment by AzureAssaulter** [comment]
I would like to add that Norton has also Flagged EDMC.exe as a trojan. It was flagged in both the installer and the location it was installed.

**Comment by Bongo982** [comment]
I tried to install EDMC but Bitdefender is running amok. 
Is it so fucking hard to make EDMC for people without a master in computer science? 
Sorry, guys but I give up, it's not worth the hassle.

**Reply by Athanasius** [reply]
This is entirely **not** our fault.

The fault lies with the Anti-Virus vendors who are not doing due diligence on their fingerprinting.  As best we can tell they're flagging certain py2exe code (the program we use to build our python programs into windows executables so that we don't have to ship a literal python runtime in our installer) as malware **because there is actual malware that also uses py2exe to package their programs**.  Rather than doing any analysis of the actual python code that gets run they just see py2exe and go "oh, malware", despite it itself being 100% harmless.

Imagine a py2exe produced executable has two parts to the file:
```
          1                                2
|--- py2exe code ----|--- embedded python 'compiled' code----|
```
The bad stuff in actual malware is in that second part, not the first.  But the AV vendors fingerprint the first part and then anything else with it (which is anything that uses py2exe to make an executable) also gets flagged as malware **despite the actual malware being in part 2 above**.

If you still want to make any more effort then you could check https://github.com/EDCD/EDMarketConnector/wiki/Running-from-source .

**Reply by Bongo982** [reply]
Apolgies for my rant, it was a hard day. I work in the health sector and right now we have another spike. 
I apologise.

**Reply by Athanasius** [reply]
> Apolgies for my rant, it was a hard day. I work in the health sector and right now we have another spike.
> I apologise.

Absolutely no problem.  Thanks for all you do ❤️ .
